xscen/extract.py

```diff
@@ -157,6 +169,7 @@ def extract_dataset(
     xr_combine_kwargs: dict = None,
     preprocess: Callable = None,
     resample_methods: Optional[dict] = None,
+    mask: Union[bool, xr.Dataset] = False,
```

Comment:
Could we also accept a `DataArray` ?

I don't find the "bool" option very useful, but I see that #183 would complete this feature.

Reply:
sure for DataArray.
I guess it is only useful for me now that I already have a mask that is in the catalogue.